### PR TITLE
feat: rest history with new chat - chat history threads

### PR DESCRIPTION
**Pull request by janaka**
Implements #64 

## Description

The approach taken implements the foundation for supporting multiple threads in the UI. 

- [x] Initially only exposing a New Chat button effectively resets chat.
- [ ] render New Chat button next to load chat history or maybe for now remove load chat history button

In another PR we can add UI to:
- UX to switch between threads
- functionality to set thread `topic` text based on the initial prompt.
- deleting a thread

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] Tests
- [ ] Other chores such as maintenance

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration by modifying the list below.

- [ ] Test A
- [ ] Test B
- [ ] Test C

**Test Configuration**:

Please describe the test setup. List them below as bullet points.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] The commit message follows the convention of this project

### CODE REVIEW

File: source/docq/support/store.py [hunk 1]
```diff
@@ -71,3 +72,9 @@ def get_history_table_name(type_: FeatureType) -> str:
     """Get the history table name for a feature."""
     # Note that because it's used for database table name, `lower()` is used to ensure it's all lowercase.
     return HISTORY_TABLE_NAME.format(feature=type_.name.lower())
+
+
+def get_history_thread_table_name(type_: FeatureType) -> str:
+    """Get the history table name for a feature."""
+    # Note that because it's used for database table name, `lower()` is used to ensure it's all lowercase.
+    return HISTORY_THREAD_TABLE_NAME.format(feature=type_.name.lower())
```
Comment (cwang) [inline source/docq/support/store.py:80]: Not sure if the changes made in this file are necessary. If we're using a sqlite file for personal data (which is per current design), then we can have two tables coexist in the same file.

**Review by cwang — COMMENTED** [review]
Looks good as described in the PR, i.e. leaving some of the features in the following PR. 
One comment:
- For consideration, we may want to switch to using a single sqlite database for all data including private chat history etc.; Not sure if this PR would be a good time to make that switch. Happy with this PR following whatever patterns that were set out in the beginning. 
And one question:
- Would all chat types support threads, e.g. for pass-through chat and shared space chat?

File: source/docq/support/store.py (hunk 1 above)
Comment (janaka) [reply source/docq/support/store.py:80]: Likely I've misunderstood how SQLite works. I'll have another look at the logic. I didn't think this was creating separate DB files. Pretty sure I checked `.persisted`. It is supposed to still be two DB files, one for personal and one for shared.

**Comment by janaka** [comment]
> Looks good as described in the PR, i.e. leaving some of the features in the following PR. One comment:
> 
> * For consideration, we may want to switch to using a single sqlite database for all data including private chat history etc.; > Not sure if this PR would be a good time to make that switch. Happy with this PR following whatever patterns were set > out in the beginning.

Having a little file-level separation isn't a bad thing from an ops PoV. I'm not sure of the downside of the current setup. 

>   And one question:
> * Would all chat types support threads, e.g. for pass-through chat and shared space chat?

It should do, this is one of the bits that need a little more testing. I don't think I tested pass-through.

File: source/docq/support/store.py (hunk 1 above)
Comment (janaka) [reply source/docq/support/store.py:80]: So there are only two DB files: `usage.db` and `system.db`. `get_history_thread_table_name()` is a following the same pattern as `get_history_table_name()` which creates a table per `FeatureType`: `ASK_PERSONAL`, `ASK_SHARED`, `CHAT_PRIVATE`.

**Pull request merged**